### ISSUE TITLE
Feature request: Socket activation

### ISSUE THREAD

**Issue by brejoc** [Supervisor/supervisor#766]
I would be nice, to be able to define a port that supervisor listens on. Upon a request supervisor checks, if the service is already running or needs to be started and then hands the request to the service. After e.g. 4 hours of inactivity (or some other indicator) the service then gets terminated again.

**Comment by mnaberez** [comment]
`supervisord` does have a socket manager in the form of [`fcgi-program`](http://supervisord.org/configuration.html#fcgi-program-x-section-settings).  Despite the name, it is not tied to FastCGI and can be used with other protocols with no special configuration.  

The socket manager will start listening when any process in an `fcgi-program` group is started.  It will stop listening when the last process in an `fcgi-program` group exits.

The socket manager does not have the activation behavior described in this ticket, so I will leave this open as a feature request.

**Comment by michaelsalmon** [comment]
I think that you have the logic reversed there. Supervisord will start listening when the socket is created and then pass the socket to the spawned processes. This is pretty much how inetd et al. work. The documentation is a bit unclear on whether the spawned process needs to listen or accept the incoming connection.

**Comment by mnaberez** [comment]
> I think that you have the logic reversed there.

No, it works as I described above.  `supervisord` will create/bind/listen the socket when the first process in an `[fcgi-program]` group is started and will close the socket when the last process in that group exits.  ([source code](https://github.com/Supervisor/supervisor/blob/3.3.0/supervisor/process.py#L653-L680)).

**Comment by michaelsalmon** [comment]
I think that you have saved me some time.

I was also looking for activating a process on socket access but if supervisord starts the process before it is needed the it is useless to me. I guess that it's back to xinetd or circus then.
